### ISSUE TITLE
Can't connect to an event whose handler type is inaccessible

### ISSUE THREAD

**Issue by EntenKoeniq** [microsoft/ClearScript#365]
Hello, i'm trying to use ClearScript with my internal class and everything is working fine, but i can't use a public delegate inside this class because i'm getting an error message with something like:  "Error: Access to "Microsoft.ClearScript.EventSource<PROJECT.ScriptContext.EmptyEvent>.connect(object)" is not possible due to the protection level.".

_I'm not getting an exception using System.EventHandler instead of a custom delegate_

C# code:
```C#
...
V8ScriptEngine engine = new V8ScriptEngine();

engine.AddHostObject("API", HostItemFlags.PrivateAccess, new ScriptContext());
...

internal class ScriptContext
{
    public delegate void EmptyEvent();

    public event EmptyEvent OnTest;

    internal void OnTestInvoke()
    {
        OnTest?.Invoke();
    }
}
```

Javascript code:
```Javascript
API.OnTest.connect(function() {

});
```

Maybe i'm doing something wrong but i can't find a solution for it.

Thanks!

**Comment by ClearScriptLib** [comment]
Hi @EntenKoeniq,

We've reproduced the issue. It's an interesting one. Thanks for reporting it!

The problem here is that `ScriptContext.EmptyEvent` is inaccessible to script code.

Exposing a host object only provides access to the object's _instance_ members. `HostItemFlags.PrivateAccess` doesn't change that; it only affects which instance members are visible.

In this case, script code can see `API.OnTest` and `API.OnTestInvoke`, but not `ScriptContext.EmptyEvent`, as that's a _static_ member – a nested type.

Nevertheless, the expression `API.OnTest` yields an instance of the class `EventSource<ScriptContext.EmptyEvent>`. Because that class is parameterized by an inaccessible type, it too is inaccessible.

This is a situation we haven't encountered before. ClearScript is behaving as intended, but in this case, we consider its behavior undesirable and therefore a bug.

We'll take a closer look at this issue and include a fix in the next release. In the meantime, one possibility is to use [`ScriptEngine.AccessContext`](https://microsoft.github.io/ClearScript/Reference/html/P_Microsoft_ClearScript_ScriptEngine_AccessContext.htm) to make all your internals visible to script code. For example, you might do something like this:

``` C#
engine.AccessContext = typeof(ScriptContext);
engine.AddHostObject("API", new ScriptContext());
```

Note that `ScriptEngine.AccessContext` only affects the visibility of exposed internals; it doesn't expose anything by itself.

Thanks again, and good luck!

**Comment by ClearScriptLib** [comment]
Fixed in [Version 7.2.5](https://github.com/microsoft/ClearScript/releases/tag/7.2.5).
